notes.md

```diff
@@ -190,6 +190,57 @@ For (possibly) Laced Data
 | 0x00        | **MUST** | Consecutive laced frames |
 Table: Lace data after lace sizes{#blockLacedData}
 
+
+## Random Access Points
+
+Random Access Points (RAP) are positions where the parser can seek to and start playback without decoding
+of what was before. In Matroska `BlockGroups` and `SimpleBlocks` can be RAPs.
+To seek to these elements it is still necessary to seek to the `Cluster` containing them
+and start playback from the `BlockGroup` or `SimpleBlock` that is a RAP.
+
+Because a Matroska File is usually composed of multiple tracks playing at the same time
+-- video, audio and subtitles -- to seek properly to a RAP, each selected track must be
+taken in account. Usually all audio and subtitle `BlockGroup` or `SimpleBlock` are RAP.
+They are independent of each other and can be played randomly.
+
+Video tracks on the other hand often use references to previous and future frames for better
+coding efficiency. Frames with such reference **MUST** either contain one or more
+`ReferenceBlock` Elements in their `BlockGroup` or **MUST** be marked
+as non-keyframe in a `SimpleBlock`; see (#simpleblock-header-flags).
+
+Frames that are RAP -- i.e. they don't depend on other frames -- **MUST** set the keyframe
+flag if they are in a `SimpleBlock` or their parent `BlockGroup` **MUST NOT** contain
+any `ReferenceBlock`.
+
+For legacy reasons, audio and subtitle track **MAY** use a `SimpleBlock` without the keyframe
+flag set, even though they are RAPs. So for audio and subtitle tracks, the only way to specify
+a frame is not a RAP is to put it in a `BlockGroup` with a `ReferenceBlock`. In all other cases
+they **MUST** be considered as decodable on their own.
+
+When a frame in a `BlockGroup` is not a RAP, all references **SHOULD** be listed as a `ReferenceBlock`,
+at least some of them, even not accurate, or one `ReferenceBlock` with the value "0" (self/unknown reference).
+
+There may be cases where the use of `BlockGroup` is necessary, as the frame may need a
+`BlockDuration`, `BlockAdditions`, `CodecState` or a `DiscardPadding` element.
+For thoses cases a `SimpleBlock` **MUST NOT** be used, the reference information **SHOULD** be recovered.
+The lack of `ReferenceBlock` would mean all frames are RAP and seeking on
+frames that actually depend on other frames **MAY** create bogus output or even crash.
+
+Intra-only video frames, such as the ones found in AV1, can be decoded without any other
+frame, but they don't reset the codec state. So seeking to these frames is not possible
+as the next frames may need frames that are not known from this seeking point.
+Such intre-only frames **MUST NOT** be considered as keyframes so the keyframe flag
+**MUST NOT** be set in the `SimpleBlock` or a `ReferenceBlock` **MUST** be used
+to signify the frame is not a RAP. The timestamp value of the `ReferenceBlock` **MUST**
+be "0", meaning it's referencing itself.
```

Comment:
This last change, especially the normative **MUST**, invalidates existing files. Being very aggressive here. What for?

Reply:
typo `intra-only`

This is a very particular case that only exist in AV1 yet AFAIK. The value "0" seems to be the correct/logical way. ie in a system that would interpret `ReferenceBlock` properly, it could cut the file accurately, keep that frame and also marking it a intra-only. Any other value (in the past) would mean it's referencing a frame which would need to be kept (and all other frames it depends on) when in fact it's not the case.

IMO for new features we should enforce clean rules, rather than

> listed as a `ReferenceBlock`, at least some of them, even not accurate

Also if we replace it with a **SHOULD** we have to explain why/when another could be used, but what would be the reason not to set 0 ?

Reply:
Updated the paragraph:
* fixed typo
* added VP9 to codecs supporting intra-only frames

In libavcodec an MPEG video stream is also considered as intra-only when the GOP size is 1 (or 0).

Reply:
For `Rolling/Periodic Intra Refresh` mentioned in #419 (ie there's no actual keyframe, the keyframe parts are spread over frames, periodically) this is possible to tell the matroska demuxer how much data it needs to seek before the seeking point desired using `SeekPreRoll`. A track using `Periodic Intra Refresh` may have no actual keyframe, maybe not even the first frame. But with the `SeekPreRoll` it should be possible to set each frame as a keyframe or RAP, the frames will be fully usable only after `SeekPreRoll` has passed.

This is unrelated to intra-only which are decoded on their own but subsequent frames may depend on older frames. So they can't be considered RAP. Hence the use of `ReferenceBlock` with a special value, which is already defined as:

> The value "0" **MAY** also be used to signify this `Block` cannot be decoded on its own, but without knownledge of which `Block` is necessary. In this case, other `ReferenceBlock` **MUST NOT** be found in the same `BlockGroup`.

So this paragraph just expresses this in terms of RAP.